src/Npgsql/NpgsqlConnection.cs

```diff
@@ -934,15 +934,23 @@ protected override void Dispose(bool disposing)
         /// Releases all resources used by the <see cref="NpgsqlConnection"/>.
         /// </summary>
 #if NETSTANDARD2_0
-        public async ValueTask DisposeAsync()
+        public ValueTask DisposeAsync()
 #else
-        public override async ValueTask DisposeAsync()
+        public override ValueTask DisposeAsync()
 #endif
         {
-            if (_disposed)
-                return;
-            await CloseAsync();
-            _disposed = true;
+            using (NoSynchronizationContextScope.Enter())
+                return DisposeAsyncCore();
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
FWIW I don't this this has any effect on async methods - I don't think the JIT inlines them (but not sure, we can check. Also no harm done and who knows, it could happen in the future).

Reply:
Just copy-pasted this from `NpgsqlDataReader.DisposeAsync`.